daemon/graphdriver/overlay2/overlay.go

```diff
@@ -191,11 +192,11 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		// Try to enable project quota support over xfs.
 		if d.quotaCtl, err = quota.NewControl(home); err == nil {
 			projectQuotaSupported = true
+		} else if opts.quota.Size > 0 {
+			return nil, fmt.Errorf("Project quota is not supported: %v", err)
```

Comment:
Oh, I see, here is the check and error out if quota is not supported. But this will run only if we are running on to of xfs.

How about moving this check out of backingFs == "xfs" if block.

Also error message could probably be more verbose. Say.

If we are not running on xfs, that could be first check and error message could be.

`Storage option overlay2.size not supported. Underlying filesystem needs to be xfs`

And if we are running on xfs but not supporting quota, we could say.

`Storage option overlay2.size not supported. Filesystem is does not support project quota`